ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/TransactionConversion.scala

```diff
@@ -146,6 +146,8 @@ trait TransactionConversion {
       },
       if (includeParentWitnesses) create.witnesses
       else create.stakeholders,
```

Comment:
I think we are silently introducing a breaking change here. Before this PR, `create.stakeholders` was already filtered up to witnesses, but now it includes all `signatories` and `observers`.

Reply:
This is true, thanks for bringing this up. I was originally intersecting both `signatories` and `observers` with the witnesses. This lead to two (already existing) conformance tests failing, which were actually fixed by me no longing intersecting the parties with the witnesses. I don't know why that would be. I'll try to run some further tests and see if restoring the `stakeholder`s to their previous state works. Still, it would be unclear to me why those tests would fail, as I would expect the union of signatories and observers intersected with witnesses to be the same as stakeholders intersected with witnesses, unless I'm missing something in how stakeholders are interpreted from the interpreter.

Note: it looks like there is some mismatch between what the interpreter and what the API mean by _witness_. It looks like the former means "anyone who has visibility on a given event", whereas the Ledger API documentation says If anything is not clear let's talk about it better.

Reply:
The field `stakeholders` as it is used right now in master is a bit misleading, because it was already filtered up to `witnesses`, so the statement `stakeholders = signatories union observers` was never true in this particular case.

Reply:
[Here](https://github.com/digital-asset/daml/blob/6ba26f4e1f9e987f2f802ad494a94c55edd2b003/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Event.scala#L159) is where the initial "filtering" happens.

Reply:
This is what confuses me. Before when populating signatories and observers I intersected them with witnesses, which I would have assumed would have made the union valid and non-breaking as compared to the previous implementation. But this caused two conformance tests to fail that have been fixed by removing the intersection with witnesses. I'm running another test filtering stakeholders up to the witnesses as before, but I don't see why my previous approach wasn't working.

Reply:
Mine was simply a logic error. If that is ok with you I'd be addressing this comment by restoring `stakeholders` to their previous state (by construction) while having unfiltered `signatories` and `observers`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/1814/commits/83b6e82e2e24697e55b7cfd77110e5e034391e0c

Reply:
So the problem with this is that the relationship `stakeholders == signatories union observers` doesn't hold anymore.

Reply:
Since this restores the previous behavior as asked from your review, I assume what you mean is that you'd like both `signatories` and `observers` to be filtered up to the witnesses just like `stakeholders` are, right?

Reply:
Me and @gerolf-da briefly talked about this yesterday, and we agreed that

1. It's a relatively "light" breaking change, especially assuming that most readers treat the list of signatories as a set;
2. It makes much more sense to not filter the signatories. The witnesses already serve the purpose of "why am I seeing this?". The signatories are a property of the contract, and thus should be left intact.

Reply:
Ok, then I would revert https://github.com/digital-asset/daml/commit/5c6810dc326733273b2c6e78c4766979000fb503 to restore the unfiltered behavior, is that ok @gerolf-da?

Reply:
Reverted by https://github.com/digital-asset/daml/pull/1814/commits/736ca823683a61c9a1b3c1d70b4dfd18c1d395c9